File: python/src/loadstrike_sample_reference/features/transport_http/http_load_shared_client_example.py
from __future__ import annotations

from loadstrike_sdk import (
    LoadStrikeRunner,
    LoadStrikeScenario,
    LoadStrikeSimulation,
    LoadStrikeStep,
)

from .http_load_shared_client_support import close_shared_session, post_order


def build():
    scenario = (
        LoadStrikeScenario.create("http-load-shared-client", _submit_orders)
        .with_load_simulations(LoadStrikeSimulation.inject(25, 1, 30))
    )

    try:
        result = (
            LoadStrikeRunner.register_scenarios(scenario)
            .with_runner_key("runner_dummy_orders_reference")
            .with_test_suite("orders-reference")
            .with_test_name("http-load-shared-client")
            .without_reports()
            .run()
        )
        return {
            "note": "Close the shared HTTP session when the run ends.",
            "Result": result,
        }
    finally:
        close_shared_session()


def _submit_orders(context):
    context.scenario_instance_data.setdefault("tenantId", "tenant-a")
    order_id = f"ord-{context.invocation_number}"

    return LoadStrikeStep.run(
        "POST /orders",
        context,
        lambda: post_order(order_id, context.scenario_instance_data["tenantId"]),
    ).as_reply()

File: js/src/features/transport_http/http-load-shared-client-example.mjs
import {
  LoadStrikeRunner,
  LoadStrikeScenario,
  LoadStrikeSimulation,
  LoadStrikeStep,
} from "@loadstrike/loadstrike-sdk";
import {
  closeSharedHttpClient,
  postOrder,
} from "./http-load-shared-client-support.mjs";

export async function build() {
  const scenario = LoadStrikeScenario
    .create("http-load-shared-client", async (context) => {
      context.scenarioInstanceData.tenantId ??= "tenant-a";
      const orderId = `ord-${context.invocationNumber}`;

      const step = await LoadStrikeStep.run("POST /orders", context, async () =>
        postOrder(orderId, String(context.scenarioInstanceData.tenantId)));

      return step.asReply();
    })
    .withLoadSimulations(LoadStrikeSimulation.inject(25, 1, 30));

  try {
    const result = await LoadStrikeRunner
      .registerScenarios(scenario)
      .withRunnerKey("runner_dummy_orders_reference")
      .withTestSuite("orders-reference")
      .withTestName("http-load-shared-client")
      .withoutReports()
      .run();

    return {
      note: "Close the shared HTTP client when the run ends.",
      result,
    };
  } finally {
    await closeSharedHttpClient();
  }
}
